### PR TITLE
Add PullRequestMonitorHandlers::MergeTargetTitler

### LINKED ISSUE #174: PRs made to non-master branch should be labeled
We can use labels, or we can modify the PR title to say the branch name.

### PR DESCRIPTION
**Pull request by Fryguy**
which will change a PR title to tag it with the target branch if it is
not master.

Example:
  PR titled "My PR title" is created targeting branch "foo", so PR title
  will be changed to "[FOO] My PR title"

Closes #174

In order to support this new type of handler, this PR adds reacting to PR changes and calling dedicated handlers.

@bdunne Please review.

### CODE REVIEW

**Comment by Fryguy** [comment]
cc @chessbyte ...once this is merged and deployed, you will not need to manually mark PRs as `[DARGA]`

File: app/workers/pull_request_monitor_handlers/merge_target_titler.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        :user   => github.user,
+        :repo   => github.repo,
+        :number => pr_number,
+        :title  => new_pr_title
+      )
+    end
+  end
+
+  def new_pr_title
+    "[#{merge_target.upcase}] #{pr_title}"
```
Comment (bdunne) [inline app/workers/pull_request_monitor_handlers/merge_target_titler.rb]: Do you need to truncate the string at a max length?

File: spec/workers/pull_request_monitor_handlers/merge_target_titler_spec.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    stub_settings(:merge_target_titler => {:enabled_repos => [branch.repo.name]})
+  end
+
+  context "when the branch has a non-master merge target" do
+    before do
+      branch.update_attributes!(:merge_target => "darga")
+    end
+
+    it "and not already titled" do
+      expect(github_service).to receive(:user).and_return("SomeUser")
```
Comment (bdunne) [inline spec/workers/pull_request_monitor_handlers/merge_target_titler_spec.rb]: wasn't expecting this...

File: app/workers/pull_request_monitor_handlers/merge_target_titler.rb (hunk 1 above)
Comment (bdunne) [inline app/workers/pull_request_monitor_handlers/merge_target_titler.rb]: Do we need to worry about other title workers changing the title at the same time?

Comment (Fryguy) [inline app/workers/pull_request_monitor_handlers/merge_target_titler.rb]: I'll check on the length...good question on the concurrency aspect.  I may need a helper method like add_pr_title_tag that is wrapped in a mutex.

File: spec/workers/pull_request_monitor_handlers/merge_target_titler_spec.rb (hunk 2 above)
Comment (Fryguy) [inline spec/workers/pull_request_monitor_handlers/merge_target_titler_spec.rb]: Yeah, for some reason the github api requires that I "resend" the user and repo...it's not being honored from the higher level calls.  It feels like a bug in the gem.

**Pull request merged**